componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionActivity.java

```diff
@@ -351,13 +345,28 @@ public boolean onOptionsItemSelected(MenuItem item) {
     @Override
     public void onProceedToAnalysisScreen(@NonNull Document document) {
         LOG.debug("Proceed to Analysis Screen with document {}", document);
-        // As the library requests us to go to the Analysis Screen we should only remove the
-        // listener.
-        // We should not cancel the analysis here as we don't know, if we proceed because the
-        // analysis didn't complete or
-        // the user rotated the image
-        getSingleDocumentAnalyzer().removeListener();
-        showFragment(getAnalysisFragment(document));
+        // If we haven't received any valid extractions we query the Gini Vision Library
+        // whether we should show the the Gini Vision No Results Screen
+        if (mNoExtractionsFound && GiniVisionCoordinator.shouldShowGiniVisionNoResultsScreen(document)) {
```

Comment:
`onProceedToAnalysisScreen` is called when the user pressed Next in the Review Screen before analysis was reported as finished (with `ReviewFragment#onDocumentAnalyzed()`).

We have to use the `mNoExtractionsFound` bool flag to know if our analysis finished w/o valid results and we should try to show the No Results Screen.